src/com/TTT/BoardPrinter.java

```diff
@@ -10,11 +10,19 @@ public BoardPrinter(PrintStream out) {
     }
 
     public void printBoard(Board board) {
-        for(int i = 0; i < board.size(); i++)
+        int rowSize = 3;
```

Comment:
Lots of magic numbers in here, with just the one extracted out. I'd suggest having a think about how you may be able to simplify this code. I would be very hesitant to change this code due to its complexity and differing levels of abstraction. Could use some effective refactoring to make it clear what's going on (I don't want to think about the maths in the main algorithm for printing the board. It should read clearly, with those details hidden away in methods that express the intent of each calculation / operation.